api/v1beta1/hetznerbaremetalmachine_types.go

```diff
@@ -76,85 +76,102 @@ type HetznerBareMetalMachineSpec struct {
 type HostSelectorRequirement struct {
-	Key      string             `json:"key"`
+	// Key defines the key of the label that should be matched in the host object.
+	Key string `json:"key"`
+
+	// Operator defines the [selection operator](https://pkg.go.dev/k8s.io/apimachinery@v0.23.4/pkg/selection?utm_source=gopls#Operator).
 	Operator selection.Operator `json:"operator"`
-	Values   []string           `json:"values"`
+
+	// Values define the values whose relation to the label value in the host machine is defined by the selection operator.
+	Values []string `json:"values"`
 }
 
 // SSHSpec defines specs for SSH.
 type SSHSpec struct {
-	// SecretRef gives reference to the secret.
+	// SecretRef gives reference to the secret where the SSH key is stored.
 	SecretRef SSHSecretRef `json:"secretRef"`
 
-	// PortAfterInstallImage specifies the port that has to be used to connect to the machine after install image.
+	// PortAfterInstallImage specifies the port that has to be used to connect to the machine
+	// by reaching the server via SSH after installing the image successfully.
 	// +kubebuilder:default=22
 	// +optional
 	PortAfterInstallImage int `json:"portAfterInstallImage"`
 
-	// PortAfterCloudInit specifies the port that has to be used to connect to the machine after cloud init.
+	// PortAfterCloudInit specifies the port that has to be used to connect to the machine
+	// by reaching the server via SSH after the successful completion of cloud init.
 	// +optional
 	PortAfterCloudInit int `json:"portAfterCloudInit"`
 }
 
-// SSHSecretRef defines the secret containing all information of the SSH key used for Hetzner robot.
+// SSHSecretRef defines the secret containing all information of the SSH key used for the Hetzner robot.
 type SSHSecretRef struct {
-	Name string          `json:"name"`
-	Key  SSHSecretKeyRef `json:"key"`
+	// Name is the name of the secret.
+	Name string `json:"name"`
+
+	// Key contains details about the keys used in the data of the secret.
+	Key SSHSecretKeyRef `json:"key"`
 }
 
 // SSHSecretKeyRef defines the key name of the SSHSecret.
 type SSHSecretKeyRef struct {
-	Name       string `json:"name"`
-	PublicKey  string `json:"publicKey"`
+	// Name is the key in the secret's data where the SSH key's name is stored.
+	Name string `json:"name"`
+
+	// PublicKey is the key in the secret's data where the SSH key's public key is stored.
+	PublicKey string `json:"publicKey"`
+
+	// PrivateKey is the key in the secret's data where the SSH key's private key is stored.
 	PrivateKey string `json:"privateKey"`
 }
 
 // InstallImage defines the configuration for InstallImage.
 type InstallImage struct {
-	// Image is the image to be provisioned.
+	// Image is the image to be provisioned. It defines the image
+	// for baremetal machine. Find more details on the image [here](https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/reference/hetzner-bare-metal-machine-template.md#installimageimage).
```

Comment:
this is very dangerous because we will change this reference section in the docs. We will also forget this here if we do ;D can you just reference the docs without any sub-path to the reference section?